### PR TITLE
M2-2603: dataviz - subscales: wrong calculation fix

### PR DESCRIPTION
**Pull request by nmialik**

### CODE REVIEW

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/BarChart/BarChart.tsx [hunk 1]
```diff
@@ -8,15 +8,19 @@ import {
...
 
 import { getDatasets } from './BarChart.utils';
 import { BarChartProps, CustomLegend } from './BarChart.types';
-import { BAR_CHART_LABEL_WIDTH_Y } from '../Charts.const';
+import { SUBSCALES_CHART_LABEL_WIDTH_Y } from '../Charts.const';
 
 ChartJS.register(CategoryScale, LinearScale, BarElement, Tooltip, Legend);
 
 export const BarChart = ({ chartData }: BarChartProps) => {
+  const responseScores = chartData.map(({ score }) => score);
```
Comment (muhurov) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/BarChart/BarChart.tsx:21]: you can use pluck from share/utils

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/LineChart/LineChart.utils.ts [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
 
-export const getOptions = (lang: keyof typeof locales, minDate: Date, maxDate: Date) => {
+export const getOptions = (
+  lang: keyof typeof locales,
+  minDate: Date,
+  maxDate: Date,
+  data: SubscaleChartData,
+) => {
+  const responses = data.subscales.map((subscale) => subscale.activityCompletions);
+  const responsesScores = responses.flat().map((response) => response.score);
```
Comment (muhurov) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/LineChart/LineChart.utils.ts:19]: pluck

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/AllScores/AllScores.styles.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import styled from '@emotion/styled';
```
[the remark this answers is not in the document]
Comment (muhurov) [reply src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/AllScores/AllScores.styles.ts:1]:
```suggestion
import { styled, Box } from '@mui/material';
```

**Comment by alex-i-e** [comment]
pls take a look on failed checks

**Review by alex-i-e — APPROVED** [review]

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**